crates/primitives/src/sorted_vec.rs

```diff
@@ -0,0 +1,194 @@
+use std::cmp::Ordering;
+
+use borsh::{BorshDeserialize, BorshSerialize};
+
+/// A vector wrapper that ensures the elements are sorted
+#[derive(Debug, Clone, BorshSerialize, BorshDeserialize)]
+pub struct SortedVec<T> {
+    inner: Vec<T>,
+}
+
+impl<T: Ord + Clone> SortedVec<T> {
+    /// Creates a new, empty `SortedVec`.
+    pub fn new() -> Self {
+        Self { inner: Vec::new() }
+    }
```

Comment:
We should also do an [`pub fn with_capacity`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.with_capacity).